libraries/AC_AttitudeControl/AC_AttitudeControl.cpp

```diff
@@ -745,6 +765,18 @@ Quaternion AC_AttitudeControl::attitude_from_thrust_vector(Vector3f thrust_vecto
     return thrust_vec_quat*yaw_quat;
 }
 
+// Calculates the body frame angular velocities to follow the target attitude
+void AC_AttitudeControl::update_attitude_target()
+{
+    // rotate target and normalize
+    Quaternion attitude_target_update;
+    attitude_target_update.from_axis_angle(Vector3f{_ang_vel_target.x * _dt, _ang_vel_target.y * _dt, _ang_vel_target.z * _dt});
```

Comment:
Given other changes coming I think it would be better to take a copy of the attitude target, do the update and then finally assign back from the local variable. Might also give the optimizer better opportunites.

Reply:
`attitude_target = _attitude_target;
attitude_target = attitude_target * attitude_target_update;
attitude_target.normalize();
_attitude_target = attitude_target;`

You mean this right?

Reply:
yeah

Reply:
i think the _dt multiply could be outside the Vector3f{}

Reply:
if you do it like this:
```
    Quaternion attitude_target_update, att = _attitude_target;
    attitude_target_update.from_axis_angle(_ang_vel_target * _dt);
    att *= attitude_target_update;
    
    // ensure Quaternion stay normalised
    att.normalize();
```
it saves 12 bytes